util/topgen/merge.py

```diff
@@ -173,6 +173,11 @@ def elaborate_instance(instance, block: IpBlock):
         if isinstance(s['width'], Parameter):
             for p in instance["param_list"]:
                 if p['name'] == s['width'].name:
+                    # When mangeling the name, we first need to deep copy the param. Otherwise, we
+                    # reference always the same single parameter. If there are multiple instances
+                    # having the same parameter, we would overwrite the previously mangled param
+                    # of another instance
+                    s['width'] = deepcopy(s['width'])
                     s['width'].name_top = p['name_top']
```

Comment:
Hmm, I'm a bit confused about this. I don't think that `Parameter` has a `name_top` member. Not necessarily for this PR, but would you mind tidying this up?

Reply:
Add that parameter to the `Parameter` class.